### PR TITLE
add VQVAE from PaintMind

### PR DESCRIPTION
**Pull request by neggles**
adds the VQGAN VAE from PaintMind and a test script

### CODE REVIEW

**Comment by korakoe** [comment]
Awesome!

**Review by korakoe — APPROVED** [review]

File: scripts/vqvae_test.py [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+image_dir = Path.cwd().joinpath("temp")
+image_dir.mkdir(exist_ok=True, parents=True)
+
+# image transforms for the VQVAE
+transform_enc = T.Compose([T.Resize(512), T.RandomCrop(256), T.ToTensor()])
+transform_dec = T.Compose([T.ConvertImageDtype(torch.uint8), T.ToPILImage()])
+
+
+def get_save_path(path: Path, append: str) -> Path:
+    return path.with_stem(f"{path.stem}-{append}")
```
Comment (ZeroCool940711) [inline scripts/vqvae_test.py]: It seems like `path.with_stem` will only work on python 3.9 and higher, replacing it with 
```
stem = f"{path.stem}-{append}{path.suffix}"
return path.with_name(f'{path.stem}_{stem}{path.suffix}')
```
so it works on older versions.

File: scripts/vqvae_test.py [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+
+    # load VAE
+    logger.info(f"Loading VQVAE from {model_repo}/{model_subdir}...")
+    vae: VQVAE = VQVAE.from_pretrained(model_repo, subfolder=model_subdir, torch_dtype=dtype)
+    vae = vae.to(torch_device)
+    logger.info(f"Loaded VQVAE from {model_repo} to {vae.device} with dtype {vae.dtype}")
+
+    # download and process images
+    for image in test_images:
+        image_path = hf_hub_download(model_repo, subfolder="images", filename=image, output_dir=image_dir)
```
Comment (ZeroCool940711) [inline scripts/vqvae_test.py]: On the latest version of the hf_hub the `output_dir` is gone, we should use `local_dir` instead.

File: scripts/vqvae_test.py [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+        image_prepro = transform_dec(image_tensor)
+        image_recon = transform_dec(reconstructed)
+        canvas = Image.new("RGB", (512 + 12, 256 + 8), (248, 248, 242))
+        canvas.paste(image_prepro, (4, 4))
+        canvas.paste(image_recon, (256 + 8, 4))
+        save_path = get_save_path(image_path, "compare")
+        canvas.save(save_path)
+        logger.info(f"Saved comparison image to {save_path}")
+
+    logger.info("Done!")
```
Comment (ZeroCool940711) [inline scripts/vqvae_test.py:80]: Call `main()` at the end of the script so we can run it standalone for testing.

File: scripts/vqvae_test.py (hunk 2 above)
Comment (neggles) [reply scripts/vqvae_test.py]: Ah yep. I blame GitHub Copilot 😅 will fix

File: scripts/vqvae_test.py (hunk 3 above)
Comment (neggles) [reply scripts/vqvae_test.py:80]: You can tell i wrote this at like 10:30pm 😅

File: scripts/vqvae_test.py (hunk 1 above)
Comment (neggles) [reply scripts/vqvae_test.py]: of all things...

**Comment by neggles** [comment]
Fixed and tested, should be ready for rebase-and-merge 😄

**Comment by neggles** [comment]
OK, let me just make a couple small tweaks to some names...

**Pull request merged**